pkg/cryptutil/certificates_index.go

```diff
@@ -32,6 +32,14 @@ func (c *CertificatesIndex) Add(cert *x509.Certificate) {
 	}
 }
 
+// Delete deletes a certificate from the index.
+func (c *CertificatesIndex) Delete(cert *x509.Certificate) {
+	usage := getCertUsage(cert)
+	for _, name := range cert.DNSNames {
+		c.delete(name, usage)
+	}
+}
+
```

Comment:
Would the CertificatesIndex ever need mutex locking? hard to tell from usages if it could be concurrently accessed/modified.

Reply:
that is out of scope for this particular component

Reply:
I only ask because before it was add-only, which can race but won't blow up your program; on the other hand, concurrent add/delete is a runtime panic. this may or may not actually be possible, but it was hard to tell